libs/src/api/serviceProvider.js

```diff
@@ -110,10 +110,9 @@ class ServiceProvider extends Base {
     // Primary: select the lowest-latency
     const primary = timings[0] ? timings[0].request.id : null
 
-    // Secondaries: select randomly
     // TODO: Implement geolocation-based selection
-    const secondaries = sampleSize(timings.slice(1), numberOfNodes - 1)
-      .map(timing => timing.request.id)
+    // Secondaries: Timings will be sorted according to highest version.
+    const secondaries = [timings[1] ? timings[1].request.id : null, timings[2] ? timings[2].request.id : null]
```

Comment:
I think it's ok for secondaries to not necessarily be highest version. You can probably leave sample size. This change here also had the side effect that you could end up with
[null, null]

Reply:
you can actually end up with [null, null] in the original method too if the timing array is less than a size of 3. this one just chooses one sequentially, while the other picks 2 random ones

Reply:
👍  ok will update